### DISCUSSION TITLE
How to get signing key from HD wallet?

### DISCUSSION THREAD

**Discussion by metawilliam** [Python-Cardano/pycardano#131] [Q&A]
I am trying to use HD Wallet. I want to get the signing key from the HD wallet object, but the "from_hdwallet" function in PaymentExtendedSigningKey seems to have issue.

The "derived_pri" is HD wallet object
It show the error log as follow:
"extended_psk = PaymentExtendedSigningKey.from_hdwallet(derived_pri)
File "../lib/python3.9/site-packages/pycardano/key.py", line 195, in from_hdwallet
    return Key(
TypeError: __init__() got an unexpected keyword argument 'type'"

May I ask how to get the signing key from the HD Wallet object?
It is okay to use the public key and get the address using HD Wallet object.

The version of pycardano using is 0.7.0

**Comment by cffls** [comment]
Hi @metawilliam , thanks for reporting this issue. Looks like this is a bug that is fixed in this commit: https://github.com/Python-Cardano/pycardano/commit/69349fba6ac94f3572fe54bd027d19117755e57a Could you please try using the latest version of pycardano in main branch? We will soon release a new pycardano version that has this patch.

**Comment by henryyuanheng-wang** [comment]
Hi @metawilliam, with the bug fixed, see example below on how to get signing key and how to get address associated:

```
hdwallet = HDWallet.from_mnemonic(YOUR_MNEMONIC_WORDS)

hdwallet_spend = hdwallet.derive_from_path("m/1852'/1815'/0'/0/0")
spend_public_key = hdwallet_spend.public_key
spend_vk = PaymentVerificationKey.from_primitive(spend_public_key)
// or you can get the following if you need extended key: 
// spend_extended_sk = ExtendedSigningKey.from_hdwallet(hdwallet_spend)
// spend_extended_vk = spend_extended_sk.to_verification_key()

hdwallet_stake = hdwallet.derive_from_path("m/1852'/1815'/0'/2/0")
stake_public_key = hdwallet_stake.public_key
stake_vk = PaymentVerificationKey.from_primitive(stake_public_key)
// or you can get the following: 
// stake_extended_sk = ExtendedSigningKey.from_hdwallet(hdwallet_stake)
// stake_extended_vk = stake_extended_sk.to_verification_key()

address = Address(spend_vk.hash(), stake_vk.hash(), network=Network.MAINNET)
// or Address(spend_extended_vk.hash(), stake_extended_vk.hash(), , network=Network.MAINNET)
```

Hope this helps!
